src/settings/__tests__/settingsReducer-test.js

```diff
@@ -121,4 +125,52 @@ describe('settingsReducer', () => {
       expect(actualState).toEqual(expectedState);
     });
   });
+
+  describe('EVENT', () => {
+    describe('type `user_settings`, op `update`', () => {
+      const eventCommon = { id: 0, type: EventTypes.user_settings, op: 'update' };
+
+      const mkCheck = <S: $Keys<SettingsState>, E: $Keys<UserSettings>>(
+        statePropertyName: S,
+        eventPropertyName: E,
+      ): ((
+        $ElementType<SettingsState, S>,
+        $ElementType<UserSettings, E>,
+        $ElementType<SettingsState, S>,
+      ) => void) => (initialStateValue, eventValue, expectedStateValue) => {
+        const initialState = { ...baseState };
+        /* $FlowFixMe[incompatible-type]: Trust that the caller passed the
+           right kind of value for its chosen key. */
+        initialState[statePropertyName] = initialStateValue;
+
+        const expectedState = { ...initialState };
+        /* $FlowFixMe[incompatible-type]: Trust that the caller passed the
+           right kind of value for its chosen key. */
+        expectedState[statePropertyName] = expectedStateValue;
+
+        expect(
+          settingsReducer(initialState, {
+            type: EVENT,
+            event: { ...eventCommon, property: eventPropertyName, value: eventValue },
+          }),
+        ).toEqual(expectedState);
+      };
+
+      describe('offlineNotification / enable_offline_push_notifications', () => {
+        const check = mkCheck('offlineNotification', 'enable_offline_push_notifications');
+        check(true, true, true);
+        check(true, false, false);
+        check(false, true, true);
+        check(false, false, false);
+      });
+
+      describe('onlineNotification / enable_online_push_notifications', () => {
+        const check = mkCheck('onlineNotification', 'enable_online_push_notifications');
+        check(true, true, true);
```

Comment:
Did you intend to cover `enable_stream_push_notifications` too? You've gone to the trouble to make this nice local test infrastructure so that more cases are easy to add :-)